### ISSUE TITLE
nolint by linter rule

### ISSUE THREAD

**Issue by peakle** [golangci/golangci-lint#2706]
### Your feature request related to a problem? Please describe.

is some case i don't want to disable all linter checks for expression by `nolint:gocritic` but only one rule `hugeParam` for example

### Describe the solution you'd like.

add feature to write nolint directive for specific rule i propose the following syntax: `nolint:linter_name:check_name` as example `nolint:gocritic:hugeParam`

### Describe alternatives you've considered.

~

**Comment by sebastien-rosset** [comment]
For the `gocritic` example, there are three possible levels: `gocritic:ruleguard:ruleguard-rule-name`.
Ruleguard is a powerful checker under `gocritic`, and within ruleguard it's possible to have hundreds of checks.

**Comment by denisvmedia** [comment]
As for `revive`, it actually supports Its own ignore comments, but there're not parsed by `golangci-lint`. I'm wondering if we can have an API (or a convention) for the linters to support so that they can themselves ignore according to the ignore rule from golangci-lint.

I can volunteer the PR, but before starting with it we need to have a clear vision on how we want to achieve it.

**Comment by ldez** [comment]
Related to #1658

**Comment by ldez** [comment]
Duplicate of #1165
